app/helpers/custom_components_helper.rb

```diff
@@ -0,0 +1,22 @@
+module CustomComponentsHelper
+  def raw_with_custom_components(stringish)
+    fragment = Nokogiri::HTML.fragment(stringish)
+
+    fragment.xpath('.//embedded-app').each do |f|
+      f.replace render_embedded_app(f)
+    end
+
+    raw(fragment.to_s)
+  end
+
+  private
+
+  def render_embedded_app(fragment)
+    @app_id = fragment['app-id']
+
+    if @app_id == 'narodenie-rodny-list'
+      @extra_attributes = fragment.attributes.except("app-id").map { |k,v| [k.to_sym, v.value] }.to_h
```

Comment:
Toto mozes napisat ako 

```
@extra_attributes = fragment.attributes.except("app-id")
```

pripadne

```
@extra_attributes = fragment.attributes.except("app-id").with_indifferent_access
```

ak chces kluce aj ako symnboly

Reply:
Toto je uzitocne, ale tento hash ma values [`Nokogiri::XML::Attr`](https://www.rubydoc.info/github/sparklemotion/nokogiri/Nokogiri/XML/Attr), takze aj tak ich musim vsetky skonvertovat na samotne hodnoty (i.e. `v.value`).

Reply:
Oki, ale BTW fungovalo by to stale :)

```ruby
@extra_attributes = fragment.attributes.except("app-id").map { |k,v| [k.to_sym, v.value] }.to_h
puts @extra_attributes[:foo]
puts fragment.attributes.except("app-id").with_indifferent_access[:foo]
```

```ruby
~/Dev/ruby/navody.digital on embedded-app!
❯ bundle exec rspec spec/helpers/custom_components_helper_spec.rb:8
Run options: include {:locations=>{"./spec/helpers/custom_components_helper_spec.rb"=>[8]}}
bar
bar
.

Finished in 0.01745 seconds (files took 2.21 seconds to load)
1 example, 0 failures
```